### PR TITLE
Add aws-sdk-cpp recipe

### PR DESCRIPTION
**Pull request by jakebolewski**

### CODE REVIEW

File: recipes/aws-sdk-cpp/build.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+mkdir build && cd build
+cmake \
+  -DCMAKE_INSTALL_PREFIX="${PREFIX}" \
+  -DBUILD_ONLY='s3;core;transfer;config' \
+  -DENABLE_TESTING=OFF \
+  -DCMAKE_BUILD_TYPE=Release ..
+make -j${CPU_COUNT}
+make install
```
Comment (ocefpaf) [inline recipes/aws-sdk-cpp/build.sh]: Is there a `make check` we can run here?

File: recipes/aws-sdk-cpp/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+build:
+ number: 0
+ skip: True  # [win]
+
+requirements:
+  build:
+    - toolchain
+    - cmake >=3.2
+    - curl
+    - openssl
```
Comment (ocefpaf) [inline recipes/aws-sdk-cpp/meta.yaml]: Can you please add the pinning for these two pinned here and below?

```
- curl >=7.44.0,<8
- openssl 1.0.*
```

PS: note that we are in the process to change how we deal with pinned dependencies,
but we move to conda-build 3 we still need to add the pins here.

File: recipes/aws-sdk-cpp/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+  run:
+    - curl
+    - openssl
+    - zlib 1.2.11
+
+test:
+  requires:
+    - gcc  # [osx]
+
```
Comment (ocefpaf) [inline recipes/aws-sdk-cpp/meta.yaml]: No need for this space.

**Comment by jakebolewski** [comment]
Hi @ocefpaf  thanks for taking the time to review!

I believe I addressed all your comments.  I re-enabled testing, there is no `make check` in the SDK so the tests are run after each component is built.

File: recipes/aws-sdk-cpp/run_test.sh [hunk 4]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+# Stop on first error
+set -e
+
+if [ "$(uname)" == "Darwin" ]; then
+    # OSX
+    g++ test.cpp -I"${PREFIX}/include" -L"${PREFIX}/lib" -laws-cpp-sdk-core -o test-core
```
Comment (isuruf) [inline recipes/aws-sdk-cpp/run_test.sh]: Change this `clang++ test.cpp -I"${PREFIX}/include" -L"${PREFIX}/lib" -Wl,-rpath,${PREFIX}/lib -laws-cpp-sdk-core -o test-core`

File: recipes/aws-sdk-cpp/run_test.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    env DYLD_LIBRARY_PATH="${PREFIX}/lib:$DYLD_LIBRARY_PATH" ./test-core
```
Comment (isuruf) [inline recipes/aws-sdk-cpp/run_test.sh]: Remove `env DYLD_LIBRARY_PATH="${PREFIX}/lib:$DYLD_LIBRARY_PATH"` after adding rpath as above

File: recipes/aws-sdk-cpp/run_test.sh [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+else
+    # Linux
+    g++ test.cpp -I"${PREFIX}/include" -L"${PREFIX}/lib${ARCH}" -laws-cpp-sdk-core -o test-core
+    env LD_LIBRARY_PATH="${PREFIX}/lib${ARCH}:$LD_LIBRARY_PATH" ./test-core
```
Comment (isuruf) [inline recipes/aws-sdk-cpp/run_test.sh]: Add rpath as above and remove `LD_LIBRARY_PATH`

File: recipes/aws-sdk-cpp/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    - openssl 1.0.*
+    - zlib 1.2.11
+  run:
+    - curl >=7.44.0,<8
+    - openssl 1.0.*
+    - zlib 1.2.11
+
+test:
+  requires:
+    - gcc  # [osx]
```
Comment (isuruf) [inline recipes/aws-sdk-cpp/meta.yaml]: Remove this

**Comment by jakebolewski** [comment]
@isuruf thanks, updated

File: recipes/aws-sdk-cpp/run_test.sh [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+
+# Stop on first error
+set -e
+
+if [ "$(uname)" == "Darwin" ]; then
+    # OSX
+    clang++ test.cpp -I"${PREFIX}/include" -L"${PREFIX}/lib" -Wl,-rpath,"${PREFIX}/lib" -laws-cpp-sdk-core -o test-core
+else
+    # Linux
+    g++ test.cpp -I"${PREFIX}/include" -L"${PREFIX}/lib${ARCH}" -Wl,-rpath,"${PREFIX}/lib${ARCH}" -laws-cpp-sdk-core -o test-core
```
Comment (isuruf) [inline recipes/aws-sdk-cpp/run_test.sh]: Add `-DCMAKE_INSTALL_LIBDIR=lib` when building and remove `${ARCH}`. We are using `${PREFIX}/lib` for all packages.

Comment (jakebolewski) [reply recipes/aws-sdk-cpp/run_test.sh]: ok, updated

File: recipes/aws-sdk-cpp/run_test.sh [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+
+# Stop on first error
+set -e
+
+if [ "$(uname)" == "Darwin" ]; then
+    # OSX
+    clang++ test.cpp -I"${PREFIX}/include" -L"${PREFIX}/lib" -Wl,-rpath,"${PREFIX}/lib" -laws-cpp-sdk-core -o test-core
+else
+    # Linux
+    g++ test.cpp -I"${PREFIX}/include" -L"${PREFIX}/lib" -Wl,-rpath,"${PREFIX}/lib$" -laws-cpp-sdk-core -o test-core
```
Comment (isuruf) [inline recipes/aws-sdk-cpp/run_test.sh]: Typo. `lib$`->`lib`

Comment (jakebolewski) [reply recipes/aws-sdk-cpp/run_test.sh]: ghaa, thanks again :)

File: recipes/aws-sdk-cpp/run_test.sh [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+# Stop on first error
+set -e
+
+if [ "$(uname)" == "Darwin" ]; then
+    # OSX
+    clang++ test.cpp -I"${PREFIX}/include" -L"${PREFIX}/lib" -Wl,-rpath,"${PREFIX}/lib" -laws-cpp-sdk-core -o test-core
+else
+    # Linux
+    g++ test.cpp -I"${PREFIX}/include" -L"${PREFIX}/lib" -Wl,-rpath,"${PREFIX}/lib" -laws-cpp-sdk-core -o test-core
+fi
```
Comment (isuruf) [inline recipes/aws-sdk-cpp/run_test.sh]: Why did you remove running the executable?

Comment (jakebolewski) [reply recipes/aws-sdk-cpp/run_test.sh]: sorry, just trying to wrap this up quickly before leaving work

added it back

**Comment by ocefpaf** [comment]
Thanks @isuruf for looking into this!
Thanks @jakebolewski for the PR!
